microsetta_public_api/repo/_beta_repo.py

```diff
@@ -0,0 +1,43 @@
+from microsetta_public_api.repo._base import DiversityRepo
+from microsetta_public_api.exceptions import UnknownID, InvalidParameter
+import numpy as np
+
+
+class BetaRepo(DiversityRepo):
+
+    def __init__(self, resources):
+        super().__init__(resources)
+        self._ids = {key: set(dm.ids) for key, dm in resources.items()}
+
+    def exists(self, sample_ids, metric):
+        distance_matrix = self._get_resource(metric)
+        if isinstance(sample_ids, str):
+            return sample_ids in distance_matrix.ids
```

Comment:
This would be `O(N)`, see [ids()](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/distance/_base.py#L201-L212). I recommend doing `return sample_ids in distance_matrix` instead, which will use the [`__contains__`](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/distance/_base.py#L780-L799) codepath with a `O(1)` test.

Reply:
See https://github.com/biocore/microsetta-public-api/pull/68/commits/de46c27d3f4dfb9921110032a8d12fb8b65db1a3